app/models/alchemy/page/page_naming.rb

```diff
@@ -99,11 +106,16 @@ def convert_url_name(name)
       end
     end
 
-    # Urlname of parent page.
-    # Returns nil, if the parent is either a language root page or the root page itself
-    def parent_urlname
-      return if parent.nil? || parent.language_root? || parent.root?
-      parent.urlname
+    def nested_url_name(value)
+      (ancestor_slugs << convert_url_name(value)).join('/')
+    end
+
+    # Slugs of all visible/non-language_root ancestors.
+    # Returns [], if there is no parent, the parent is
+    # the root page itself, or url_nesting is off.
+    def ancestor_slugs
+      return [] if parent.nil? || parent.root? || !Config.get(:url_nesting)
```

Comment:
We should change this to: `!Config.get(:url_nesting) || parent.nil? || parent.root?`, because when url nesting is disabled we could spare the calls for `parent.nil?` and `parent.root?`